specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json

```diff
@@ -994,6 +1034,48 @@
         "tableName"
       ]
     },
+    "RestServiceDataset": {
+      "x-ms-discriminator-value": "RestResource",
+      "description": "A Rest service dataset.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Dataset"
+        }
+      ],
+      "properties": {
+        "typeProperties": {
+          "description": "Properties specific to this dataset type.",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/RestServiceDatasetTypeProperties"
+        }
+      }
+    },
+    "RestServiceDatasetTypeProperties": {
+      "description": "Properties specific to this dataset type.",
+      "properties": {
+        "relativeUrl": {
+          "type": "object",
+          "description": "The relative URL to the resource that the RESTful API provides. Type: string (or Expression with resultType string)."
+        },
+        "requestMethod": {
+          "type": "object",
+          "description": "The HTTP method used to call the RESTful API. The default is GET. Type: string (or Expression with resultType string)."
+        },
+        "requestBody": {
+          "type": "object",
+          "description": "The HTTP request body to the RESTful API if requestMethod is POST. Type: string (or Expression with resultType string)."
+        },
+        "additionalHeaders": {
+          "type": "object",
+          "description": "The additional HTTP headers in the request to the RESTful API. Type: string (or Expression with resultType string)."
+        },
+        "paginationRules": {
+          "description": "The pagination rules to compose next page requests.",
```

Comment:
Can this also be an Expression? Since the type is object the model will accept any type here.
Also, please change the order of type and description to be consistent. And update the comment to indicate accepted types.

Reply:
@dazhao-msft

Reply:
It actually is a dictionary.  What does Expression look like?

Reply:
Expression something like this that customers can provide:
@addhours(trigger().outputs.windowStartTime, -2)
They are useful whne they need to generate dynamic values. You need to make sure that backend supports this before allowing it in SDK too.

Reply:
There is a way to model dictionary in swagger too if you don't need Expressions

Reply:
Yes, we don't need Expression here.

Reply:
And I want to keep the type as object here, because it leaves the flexibility to change backend in the future.

Reply:
Ok, then, just change the order of type and description.

Reply:
updated in the new iteration